frontend/src/api/api.ts

```diff
@@ -12,17 +12,25 @@ let waiters: Array<Waiter> = [];
 
 export default {
   init: async function (events: EventsProvider): Promise<void> {
-    try {
-      api = new Client(METABACKEND.URL, METABACKEND.NETWORK);
-      await api.initialize();
-      events.afterSuccessfulInitialization();
-      // Wake up sleepers waiting for API.
-      for (let i = 0; i < waiters.length; i++) {
-        waiters[i](api);
+    async function initClient({ onSuccess }) {
+      try {
+        api = new Client(METABACKEND.URL, METABACKEND.NETWORK);
+        await api.initialize();
+        events.afterSuccessfulInitialization();
+        // Wake up sleepers waiting for API.
+        for (let i = 0; i < waiters.length; i++) {
+          waiters[i](api);
+        }
+        onSuccess();
+      } catch (error) {
+        handleInitError(error, events);
       }
-    } catch (error) {
-      handleInitError(error, events);
     }
+    const interval = setInterval(function () {
+      initClient({
+        onSuccess: function () { clearInterval(interval); }
```

Comment:
And here, instead of getting called back, check the return bool for success or not.

If you want to keep using setInterval (which isn't async), you can do

```typescript
initClient.then((success)=>{
    if (success) {
       clearInterval(interval)
    }
});
```

You can also find async-friendly setInterval implementations on npm.

Reply:
done!